akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpCookie.scala

```diff
@@ -113,10 +188,54 @@ final case class HttpCookie(
   /** Java API */
   def withHttpOnly(httpOnly: Boolean): headers.HttpCookie = copy(httpOnly = httpOnly)
   /** Java API */
+  def withSameSite(sameSite: jm.headers.SameSite): headers.HttpCookie = new HttpCookie(name, value, expires, maxAge, domain, path, secure, httpOnly, extension, Option(sameSite.asScala()))
+  /** Java API */
+  def withSameSite(sameSite: Optional[jm.headers.SameSite]): headers.HttpCookie = new HttpCookie(name, value, expires, maxAge, domain, path, secure, httpOnly, extension, sameSite.asScala.map(_.asScala()))
+  /** Java API */
   def withExtension(extension: String): headers.HttpCookie = copy(extension = Some(extension))
 }
 
 object HttpCookie {
+
+  @deprecated("for binary compatibility", since = "10.2.0")
+  def apply(
```

Comment:
To reduce the source-incompatibility (now requiring 'new'), I think we should duplicate this apply method into 2: one deprecated one without default values and without sameSite, and one with default values and with sameSite. Would that work?